### PR TITLE
Style changes for small widths

### PR DESCRIPTION
**Pull request by davepagurek**
Addresses https://github.com/rsms/fb-mac-messenger/issues/219

I created a stylesheet that gets injected so that the New Message button can be moved over. I also made the minimum window size smaller because there wasn't a reason for it to be as wide as it is.

Before:
screen shot 2015-11-08 at 8 46 52 pm

After:
screen shot 2015-11-08 at 11 41 19 pm

### CODE REVIEW

**Comment by grinich** [comment]
:+1: yo @rsms merge this :smile:

File: build/MainCSSDataURL.m [hunk 1]
```diff
@@ -0,0 +1 @@
...AKICAgIC8qIE1vdmUgb3ZlciBzZWFyY2ggYmFyIHRvIG1ha2Ugcm9vbSBmb3IgYnV0dG9uICovCiAgICAuXzJ5OHkgewogICAgICAgIHBhZGRpbmctbGVmdDogNDFweCAhaW1wb3J0YW50OwogICAgfQogICAgCiAgICAvKiBNYWtlIHN1cmUgdGl0bGUgZG9lc24ndCBvdmVybGFwIHdpdGggYnV0dG9ucyAqLwogICAgLl81NzQzIHsKICAgICAgICBib3gtc2l6aW5nOiBib3JkZXItYm94OwogICAgICAgIHBhZGRpbmctbGVmdDogNDFweCAhaW1wb3J0YW50OwogICAgICAgIHBhZGRpbmctcmlnaHQ6IDQxcHggIWltcG9ydGFudDsKICAgICAgICB3aWR0aDogMTAwJSAhaW1wb3J0YW50OwogICAgICAgIGJveC1zaXppbmc6IGJvcmRlci1ib3g7CiAgICB9CiAgICBoMi5fMTd3MiB7CiAgICAgICAgb3ZlcmZsb3c6IGhpZGRlbjsKICAgICAgICB0ZXh0LW92ZXJmbG93OiBlbGxpcHNpczsKICAgICAgICB3aWR0aDogMTAwJSAhaW1wb3J0YW50OwogICAgfQogICAgCiAgICAvKiBNb3ZlIGdlYXIgaWNvbiB1cCwgcGxhY2UgcmVhZCBpbmRpY2F0b3IgdW5kZXIgZ2VhdCBpY29uICovCiAgICAuXzQtMGgsIC5fMl9hMi5pbWcgewogICAgICAgIGJhY2tncm91bmQtY29sb3I6ICNGRkY7CiAgICAgICAgYm9yZGVyOiAycHggc29saWQgI0ZGRjsKICAgICAgICBib3JkZXItcmFkaXVzOiA1MCU7CiAgICAgICAgbWFyZ2luLXRvcDotNDRweDsKICAgICAgICBtYXJnaW4tcmlnaHQ6LTJweDsKICAgIH0KICAgIAogICAgLyogQ2hhbmdlIGJvcmRlcnMgb24gc2VsZWN0ZWQgY29udmVyc2F0aW9ucyAqLwogICAgLl8xaHQyIC5fNC0waCwgLl8xaHQyIC5fMl9hMi5pbWcgewogICAgICAgIGJvcmRlci1jb2xvcjogI0Y2RjZGNjsKICAgICAgICBiYWNrZ3JvdW5kLWNvbG9yOiAjRjZGNkY2OwogICAgfQp9Cg==";
```
Comment (rsms) [inline build/MainCSSDataURL.m]: Don't include generated files in the source code. There's even a .gitignore rule that should have warned you about this file.

File: Messenger/AppDelegate.mm [hunk 2]
```diff
@@ -120,7 +121,7 @@ - (void)applicationDidFinishLaunching:(NSNotification*)notification {
   if ([[NSUserDefaults standardUserDefaults] boolForKey:@"moves-with-active-space"]) {
     _window.collectionBehavior |= NSWindowCollectionBehaviorMoveToActiveSpace;
   }
-  _window.minSize = {605,300};
+  _window.minSize = {400,300};
```
Comment (rsms) [inline Messenger/AppDelegate.mm]: Have you tested this? Last time I tried this (as the comment below suggests) the view started scrolling horizontally.

**Comment by rsms** [comment]
The css selectors using classnames in the CSS file _WILL BREAK_ — CSS classes are generated based on tree relation and changes when something in the DOM tree changes (I used to work at Facebook.)

Look at [website/app/main.js](https://github.com/rsms/fb-mac-messenger/blob/master/website/app/main.js) and you'll see that we go to great lengths to avoid relying on style classnames (i.e. findReactDOMNode, dropboxSharedLinks, etc)

Suggestion: Instead of injecting a CSS file, try to accomplish the same things through JavaScript in main.js and use a combination of findReactDOMNode and narrow query selectors (that don't include style classnames.)

**Comment by davepagurek** [comment]
alright, I'll change it to work that way. Once a node is found through Javascript, is it preferable to add a CSS class to it so that styling via a CSS file can be used for things like media queries, or should I add styling in its entirety through Javascript?

File: build/MainCSSDataURL.m (hunk 1 above)
Comment (davepagurek) [inline build/MainCSSDataURL.m]: Makes sense, I'll add an ignore rule for it

File: Messenger/AppDelegate.mm (hunk 2 above)
Comment (davepagurek) [inline Messenger/AppDelegate.mm]: I've been using this build for a few weeks now without seeing any behaviour like that, and this is on OS X 10.11. What version did you notice the scrolling on?

**Comment by rsms** [comment]
As all the media queries are used for is to set a property for a certain viewport width threshold, I'd suggest just sticking with JavaScript. If we start assigning style classnames to nodes we will end up with code that's hard to read as it will stretch across files.

**Comment by davepagurek** [comment]
Alright, so now it uses Javascript. Additional stylings can be added by calling `styleComponent(reactClass, queryResponses[, options])` where:
- `queryResponses` is an object mapping a media query to a function taking in the react element and whether or not the query matches
- `options` optionally specifies `MutationObserver` options for when to reapply styles

An example call looks like this:

``` javascript
styleComponent("MessengerMasterView", {
  "(max-width: 640px)": function(el, matches) {
    el.style.backgroundColor = matches ? "black" : null;
  }
});
```

In addition to moving the New Conversation button over, I updated the way the read receipt and thread settings icons are displayed.
Before: (the first icon is being hovered over to show the settings gear)
screen shot 2015-12-03 at 9 05 20 pm
After: (the settings gear appears over the read receipt on hover)
screen shot 2015-12-03 at 9 05 49 pm
(This is definitely moving into territory of changes that don't directly affect the app being useable and can easily break on a Facebook update, so let me know if you feel that this doesn't belong in this PR.)

**Comment by davepagurek** [comment]
Brief update, needed to update the z-index of the New Conversation button to keep it in front in small widths.

**Comment by davepagurek** [comment]
Update again: now in sync with master.

**Comment by rsms** [comment]
Nice work!

**Pull request merged**